test/pkg/environment/common/environment.go

```diff
@@ -30,29 +33,44 @@ import (
 	"github.com/onsi/gomega"
 	"github.com/samber/lo"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	serializeryaml "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	controllerruntime "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
 
 	"sigs.k8s.io/karpenter/kwok/apis/v1alpha1"
 	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
 	"sigs.k8s.io/karpenter/pkg/operator"
-	"sigs.k8s.io/karpenter/pkg/test"
 	. "sigs.k8s.io/karpenter/pkg/utils/testing" //nolint:stylecheck
 	"sigs.k8s.io/karpenter/test/pkg/debug"
 )
 
 type ContextKey string
 
-const (
-	GitRefContextKey = ContextKey("gitRef")
+const GitRefContextKey = ContextKey("gitRef")
+
+// I need to add the the default kwok nodeclass path
+// That way it's not defined in code but we use it when we initialize the nodeclass
+var (
+	//go:embed default_kowknodeclass.yaml
+	defaultNodeClass []byte
+	//go:embed default_nodepool.yaml
+	defaultNodePool []byte
```

Comment:
What's the reason you need a DefaultNodePool at all? Don't you fully control it in the context of the test and you just need to update the NodeClass to point to the CloudProvider-specific NodeClass type?

Reply:
We control changing, but we don't control the capacity available  by the cloudprovider. Mainly concerned about updating the requirements

Reply:
Makes sense. Seems reasonable.